### PR TITLE
GROVE-525: fixed issue with not searching for blank facet values

### PR DESCRIPTION
**Pull request by mariannemyers**
Bugfix for bug reported by Gabo Manuel..

### CODE REVIEW

File: grove-node-server-utils/filter.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ const queryBuilder = require('./ml-query-builder.service.js')({});
 var filter = (function() {
   var queryTextDefaultHandler = function(filter) {
     var q;
-    if (filter.constraint) {
+    if (filter.constraint !== undefined) {
```
Comment (patrickmcelwee) [inline grove-node-server-utils/filter.js]: Perhaps we should also exclude `null`. What do you think @grtjn?

Comment (patrickmcelwee) [reply grove-node-server-utils/filter.js]: If we do, we could include it everywhere

Comment (patrickmcelwee) [reply grove-node-server-utils/filter.js]: Looks like the easiest way is to use `== null` and `!= null`, which covers null and undefined: https://stackoverflow.com/a/15992131

Comment (grtjn) [reply grove-node-server-utils/filter.js]: == and != are against our lint styling. I agree on adding !== null. Best to be robust..

**Comment by grtjn** [comment]
bugfixes should go into master..

**Pull request merged**